sqlsmith: make `makeStringComparison` great again

otan opened the pull request:
Previously this basically always dropped everything except bools (the
only available scalar). I think it was intended to be otherwise.

Release note: None

otan committed:
sqlsmith: make `makeStringComparison` great again

Previously this basically always dropped everything except bools (the
only available scalar). I think it was intended to be otherwise.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/73956)
<!-- Reviewable:end -->

otan committed:
sqlsmith: make `makeIn` great again

Previously this basically always dropped everything except bools (the
only available scalar). I think it was intended to be otherwise.

Release note: None
